### PR TITLE
[FLINK-27645] Update overview / supported features page for 1.0.0

### PR DESCRIPTION
**Pull request by morhidi**
- Restructured the feature summary
- Added links for quick access
- Added Known issues and limitations

### CODE REVIEW

**Comment by morhidi** [comment]
cc @wangyang0918 @mbalassi

File: docs/content/docs/concepts/overview.md [hunk 1]
```diff
@@ -30,50 +30,52 @@ Flink Kubernetes Operator acts as a control plane to manage the complete deploym
...
+  - Automated [RBAC configuration](https://nightlies.apache.org/flink/flink-kubernetes-operator-docs-main/docs/operations/rbac/)
+  - Advanced customization techniques 
+- Up-to-date public repositories 
+  - Helm [https://downloads.apache.org/flink/flink-kubernetes-operator-1.0.0/](https://downloads.apache.org/flink/flink-kubernetes-operator-1.0.0)
+  - Docker [ghcr.io/apache/flink-kubernetes-operator](http://ghcr.io/apache/flink-kubernetes-operator)
 
 # Known issues & limitations
 
-JobResultStore Resource Leak
+## JobManager High-availability
```
Comment (wangyang0918) [inline docs/content/docs/concepts/overview.md:72]: Do we have some known limitations on JobManager high-availability?

Comment (morhidi) [reply docs/content/docs/concepts/overview.md:72]: Wanted to add that the job restarts when the leader JM goes down.

**Comment by morhidi** [comment]
There are some missing pieces, but wanted to open it on for feedback, and other parts of the documentation is still moving, so links probably need to be corrected/added.

**Review by wangyang0918 — APPROVED** [review]
+1

**Comment by wangyang0918** [comment]
Merging now. Let's create new tickets if we get more feedbacks from users.

**Pull request merged**